lib/pengine/pe_actions.c

```diff
@@ -785,8 +784,8 @@ pcmk__unpack_action_meta(pcmk_resource_t *rsc, const pcmk_node_t *node,
     // Check for pcmk_monitor_timeout
     if (pcmk_is_set(pcmk_get_ra_caps(rsc_rule_data.standard),
                     pcmk_ra_cap_fence_params)
-        && (pcmk__str_eq(action_name, PCMK_ACTION_START, pcmk__str_none)
-            || pcmk_is_probe(action_name, interval_ms))) {
+        && pcmk__str_any_of(action_name, PCMK_ACTION_START, PCMK_ACTION_MONITOR,
+                            NULL)) {
```

Comment:
See e44a6d4b2 commit message -- will the controller think the action configuration changed?

Reply:
Thanks for pointing that out. The commit message is long and I missed that while refreshing for this "easy fix"...

Sigh, almost certainly. This is probably going to be a CANTFIX, unless we decide the digest change is acceptable in order to avoid a failed stonith stop action.

If it would make any difference in our decision, I'll double-check whether this can happen with any timed-out recurring monitor (with long `pcmk_monitor_timeout`) or if it's specific to the first one somehow. If it can happen with any (which it looks like should be possible), I'm surprised no one has hit and reported this.

Reply:
That, or try to find a different way to update the controller's expected timeout, which may be considerably more complicated

Reply:
We do already have a mechanism in the fencer for updating the controller's expected timeout for fencing actions, so it could be modeled on that, but it would still likely be pretty intrusive.

Doesn't the controller have both timeout and pcmk_monitor_timeout in the graph? Maybe it could just do the override itself.

Reply:
> Doesn't the controller have both timeout and pcmk_monitor_timeout in the graph? Maybe it could just do the override itself.

IIRC we strip out `pcmk_*` options from the transition graph. I suspect a fairly straightforward fix would be to add pcmk_monitor_timeout to the graph when needed, as a special XML attribute rather than with the rest of the resource parameters (to avoid breaking the hash). The controller can then pull it out and use it instead of the usual timeout.